c#/Mandoline.Api.Examples/Core/Client/APIClient.cs

```diff
@@ -0,0 +1,391 @@
+
+        if (userAssertion.Passed)
+        {
+            foreach (User user in userAssertion.Result)
+            {
+                user.Client = this;
+            }
+        }
+
+        return userAssertion;
+    }
+
+    /// <summary>
+    /// Retrieve all databanks.
+    /// </summary>
+    /// <returns>list of databanks.</returns>
+    public async Task<Assertion<IEnumerable<Core.Client.Models.Databank>>> GetDatabanksAsync()
+    {
+        Trace.WriteLine("GetDatabanksAsync()");
+        string url = ResourceLinks.DatabankPath;
+
+        Assertion<List<DatabankDto>> responseTask = await this.PerformGet<List<DatabankDto>>(url);
+
+        if (!responseTask.Passed)
+        {
+            return Assertion<IEnumerable<Databank>>.Fail(null, responseTask.Reason, responseTask.Description);
+        }
+
+        IEnumerable<Databank> result =
+            from databank in responseTask.Result
+            select new Databank(this)
+            {
+                HasAccess = databank.HasAccess,
+                DatabankCode = databank.DatabankCode,
+                DatabankColumns = databank.DatabankColumns,
+                HasQuarterlyData = databank.HasQuarterlyData,
+                EndYear = databank.EndYear,
+                MapUrl = databank.MapUrl,
+                Name = databank.Name,
+                StartYear = databank.StartYear,
+                Url = databank.Url,
+                Trees = from tree in databank.Trees
+                        select new TreeLink(this)
+                        {
+                            Name = tree.Name,
+                            TreeCode = tree.TreeCode,
+                            Url = tree.Url,
+                        },
+            };
+
+        return Assertion<IEnumerable<Databank>>.Pass(result, responseTask.Reason);
+    }
+
+    /// <summary>
+    /// Create a new saved selection.
+    /// </summary>
+    /// <param name="selectionToCreate">selection to create.</param>
+    /// <returns>created selection.</returns>
+    public async Task<Assertion<SelectionDto>> CreateSavedSelectionAsync(SelectionDto selectionToCreate, CancellationToken cancel)
+    {
+        Trace.WriteLine("CreateSavedSelectionAsync(" + selectionToCreate.Id + ")");
+        Assertion<SelectionDto> createdSelection = await this.PerformPost<SelectionDto>(selectionToCreate, ResourceLinks.SavedSelection, cancel);
+
+        return createdSelection;
+    }
+
+    /// <summary>
+    /// Download a selection.
+    /// </summary>
+    /// <param name="selection">selection to download.</param>
+    /// <returns>dataseries.</returns>
+    public async Task<Assertion<List<DataseriesDto>>> DownloadAsync(SelectionDto selection, CancellationToken cancel, int? page = null, int? pagesize = null)
+    {
+        Trace.WriteLine("DownloadAsync(" + selection + ")");
+
+        string url;
+        if (page.HasValue && pagesize.HasValue)
```

Comment:
This is just a suggestion for the future rather than for now but the if statement could be done using a ternary operator like this:

var url = page.HasValue && pagesize.HasValue
           ? string.Format("{0}?page={1}&pagesize={2}", ResourceLinks.DownloadPath, page, pagesize)
           : ResourceLinks.DownloadPath;
           
This just saves a few lines